bpf/mock/README.md

```diff
@@ -0,0 +1,71 @@
+## Dependencies
+
```

Comment:
It'd help to add a few words here about what these dependencies are and why they are used, CMake and `cmock.rb` are new things to Cilium project itself. Some folks to use Ruby because of Vagrant, but broadly it's not a widely common dependency for Cilium developers.